pkg/arvo/app/dojo.hoon

```diff
@@ -1134,8 +1134,14 @@
   ++  he-errd                                           ::  reject update
     |=  [rev=(unit sole-edit) err=@u]  ^+  +>
     =+  red=(fall rev [%nop ~])       ::  required for error location sync
+    =?  red  =(%mor -:red)
+      =/  rel  ;;((list sole-edit) +:red)
+      =/  idx  (find `(list sole-edit)`[%del `@ud`err]~ rel)
+      ?.  ?=(^ idx)
+        red
+      [%mor (oust [+:idx (lent rel)] rel)]
```

Comment:
```suggestion
    =?  red  ?=(%mor -.red)
      =/  rel  p.red
      =/  idx  (find [%del err]~ rel)
      ?~  idx  red
      [%mor (scag u.idx rel)]
```

Couple changes for better style:
- If you `?=` to type-check in the conditional, instead of plain equality check, you get typed access to `p.red`, since the type system now knows `red` is the `%mor` case.
- Prefer `.` instead of `:` for simple sub-noun access.
- I don't think you need all these explicit typecasts in the `+find` call.
- `?~  etc` in place of `?.  ?=(^ etc)` is more idiomatic.
- `+scag` takes the first n items of a list, more idiomatic than deleting anything path the nth item.